### DISCUSSION TITLE
Is there a way to emulate Drools insertLogical concept?

### DISCUSSION THREAD

**Discussion by johannz** [evrete/evrete#35] [Q&A] [answered]
https://docs.drools.org/5.6.0.Final/drools-expert-docs/html/ch02.html#d0e1037

> After regular inserts you have to retract facts explicitly. With logical assertions, the fact that was asserted will be automatically retracted when the conditions that asserted it in the first place are no longer true. Actually, it's even cleverer then that, because it will be retracted only if there isn't any single condition that supports the logical assertion.
 
I'm working on a character generator for a RPG. If the user selects a particular race, for instance Goblin, I want to assert facts about stats and skill modifiers for that race. If they change Races, I'd like to revert those facts without having to explicitly list them again.

**Comment by andbi** [comment, marked as answer]
Hi @johannz 

Unfortunately the requested feature is too complex for a simple and lightweight rule engine like Evrete, it simply doesn't track the asserting conditions. 

However, this is the second time users have requested a feature that generally requires storing facts along with some fact-related info. I will look into it, it shouldn't be too difficult, but there's still going to be a long way toward Truth Maintenance and features like `insertLogical()`.

Regards,
Andrew
